test/filter_plugin.c

```diff
@@ -850,7 +850,7 @@ test_creating_groups_using_plugins(hid_t fid)
         char *sp = subgroup_name;
 
         sp += snprintf(subgroup_name, sizeof(subgroup_name), SUBGROUP_PREFIX);
-        sprintf(sp, "%d", i);
```

Comment:
Based on a quick reading of the code, I think these two calls should just be combined into a single `snprintf(subgroup_name, sizeof(subgroup_name), SUBGROUP_PREFIX "%d");`
Doesn't seem like `sp` is needed here.